draft/0004-user-sign-in-experience.md

```diff
@@ -0,0 +1,453 @@
+---
+Start date: 2024-05-24
+Author(s): [simeng-li]
+Revision: 1
+---
+
+# Sign-in experience and experience APIs
+
+## 1. Abstract
+
+Sign-in experience (SIE) is the process of user authentication in the Logto platform. It includes the user interface and experience APIs that allow end-users to interact with the Logto platform for identity verification and profile management.
+
+## 2. Motivation
+
+In Logto, we support a variety of modern authentication methods such as password, social login, and multi-factor authentication (MFA). Developers may require a flexible and customizable sign-in experience to meet their specific use cases.
+
+Current version of `Experience APIs` in Logto are neither well-documented nor publicly available for the developers. This document aims to provide a comprehensive overview of the SIE model in Logto and introduce a new set of `Experience APIs` that allow developers to build a customized sign-in experience for their users.
+
+### 3.1 Terminology
+
+- **Sign-in experience(SIE)**: A user's experience when interacting with Logto platform. Including the front-end user interface and the back-end API calls.
+- **Experience APIs**: A set of APIs that allow end-users to interact with the Logto for identity verification and profile management.
+- **Interaction session**: A session that is initiated when a user starts interacting with the Logto platform. The session will be terminated once the user logs out or the session expires.
```

Comment:
Actually "interaction" and "session" are different concepts in `oidc-provider`. [By default](https://github.com/panva/node-oidc-provider/blob/main/docs/README.md#ttl) interaction has 1 hour TTL while session has 14 days.

To avoid confusion, I suggest we use "interaction" without "session" in the RFC.

```suggestion
- **Interaction:** A short-lived session that starts when a user initiates the sign-in experience. By default, it has an expiration time of one hour.
```